test/libsolidity/AnalysisFramework.cpp

```diff
@@ -51,9 +51,11 @@ AnalysisFramework::parseAnalyseAndReturnError(
 )
 {
 	compiler().reset();
+	// Do not insert license if it is already present.
+	bool insertLicense = _insertLicenseAndVersionPragma && _source.find("// SPDX-License-Identifier:") == string::npos;
```

Comment:
Actually we should have a generic "prepareSources" helper for all the different tests. Will do that in a separate PR.